pkg/collector/corechecks/containers/kubelet/provider/probe/provider.go

```diff
@@ -0,0 +1,110 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubelet
+
+package probe
+
+import (
+	"github.com/prometheus/common/model"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/containers/kubelet/common"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/containers/kubelet/provider/prometheus"
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
+	"github.com/DataDog/datadog-agent/pkg/tagger/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+)
+
+// Provider provides the metrics related to data collected from the `/metrics/probes` Kubelet endpoint
+type Provider struct {
+	filter *containers.Filter
+	store  workloadmeta.Store
+	*prometheus.Provider
+}
+
+func NewProvider(filter *containers.Filter, config *common.KubeletConfig, store workloadmeta.Store) *Provider {
+	provider := &Provider{
+		filter: filter,
+		store:  store,
+	}
+
+	transformers := map[string]func(*model.Sample, sender.Sender){
+		"prober_probe_total": provider.proberProbeTotal,
+	}
+
+	scraperConfig := &prometheus.ScraperConfig{AllowNotFound: true}
+	if config.ProbesMetricsEndpoint == nil || *config.ProbesMetricsEndpoint != "" {
+		scraperConfig.Path = "/metrics/probes"
+	}
+
+	provider.Provider = prometheus.NewProvider(config, transformers, scraperConfig)
+	return provider
+}
+
+func (p *Provider) proberProbeTotal(metric *model.Sample, sender sender.Sender) {
+	metricSuffix := ""
+
+	probeType := metric.Metric["probe_type"]
+	if probeType == "Liveness" {
+		metricSuffix = "liveness_probe"
+	} else if probeType == "Readiness" {
+		metricSuffix = "readiness_probe"
+	} else if probeType == "Startup" {
+		metricSuffix = "startup_probe"
+	} else {
+		log.Debugf("Unsupported probe type %s", probeType)
+		return
+	}
+
+	result := metric.Metric["result"]
+	if result == "successful" {
+		metricSuffix += ".success.total"
+	} else if result == "failed" {
+		metricSuffix += ".failure.total"
+	} else if result == "unknown" {
+		metricSuffix += ".unknown.total"
+	} else {
+		log.Debugf("Unsupported probe result %s", result)
+		return
+	}
+
+	podUid := string(metric.Metric["pod_uid"])
```

Comment:
If it is not going to be there for one, it would not be there for all, because it is coming from the output of the /metrics/probes endpoint. If it is missing, the end user would not be able to do anything about it, and I worry that the amount of times we could run into it could result in a lot of log spam if we do log a warn.

If the desire is to have logging for such an instance, then I think having it be debug level makes more sense, so that the ability to debug missing metrics still exists. But in either case I am happy to make the change if you think that the logging volume is not that much of a concern